frontend/BoardGUI.py

```diff
@@ -37,29 +41,101 @@ def build_board(self):
         self.draw_board()
 
     def draw_piece(self, turn, position):
-        if turn == Turn.BLACK:
+        if turn == Color.BLACK:
             self.game.screen.blit(self.black_piece_image, position)
         else:
             self.game.screen.blit(self.white_piece_image, position)
 
+    def remove_piece(self, row, col):
+        # Remove the piece from the internal structure
+        if (row, col) in self.pieces_on_board:
+            del self.pieces_on_board[(row, col)]
+
+    
+    def handle_mill(self,row, col):
+        if (row, col) in self.game_manager.removable_pieces:
+            # Valid piece selected to remove
+            print(f"Removing opponent's piece at ({row}, {col})")
+            self.remove_piece(row, col)  # Remove the piece from the board
+            self.draw_board()  # Redraw the board after removal
+            status = self.game_manager.remove_piece_mill(row, col)
+            if status:
+                print(f"Removed opponent's piece at ({row}, {col})")
+        else:
+            print(f"Invalid selection. Please select a piece from: {self.game_manager.removable_pieces}")
+    
+    def handle_moment_phase(self,row, col):
+        if self.game_manager.selected_piece is None:
```

Comment:
We could break these 2 conditions out into 2 separate functions. `assign_piece_to_move()` and `move_piece()`

Reply:
so previously it was all in a single method i cut this chunk and created new function to increase readability, i don't think its good idea to make multiple methods even we can do in this case. + there is no significant benefit for that